apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -53,3 +60,24 @@ func validateRAID(r *RAIDConfig) error {
 
 	return nil
 }
+
+func validateDNSName(hostaddress string) error {
+
+	if hostaddress == "" {
+		return nil
+	}
+
+	hurl, _ := url.Parse(hostaddress)
+	host, port, _ := net.SplitHostPort(hurl.Host)
```

Comment:
@hroyrh, if port is empty `host = hurl.Host` seems ok to me. But why we are not checking error in here?

Reply:
It raises error for port empty too and as for the validity of the url, the regex checks that.